torch/backends/mkldnn/quantization/functional.py

```diff
@@ -0,0 +1,203 @@
+r""" Functional interface for MKLDNN backend (quantized)."""
+r""" Aligned with torch/nn/quantized/functional.py """
+from typing import Optional
+import torch
+from torch import Tensor
+from torch.nn.modules.utils import _pair, _triple
+from torch.nn.quantized.modules.utils import _pair_from_first
+
+# Conv
+def conv1d_mkldnn(input, weight, bias,
+                  stride=1, padding=0, dilation=1, groups=1,
+                  padding_mode='zeros',
+                  scale=1.0, zero_point=0,
+                  dtype=torch.quint8):
+    r"""
+    Applies a 1D convolution over a quantized 1D input composed of several input
+    planes.
+
+    Args:
+        input: quantized input tensor of shape :math:`(\text{minibatch} , \text{in\_channels} , iW)`
+        weight: quantized filters of shape :math:`(\text{out\_channels} , \frac{\text{in\_channels}}{\text{groups}} , iW)`
+        bias: **non-quantized** bias tensor of shape :math:`(\text{out\_channels})`. The tensor type must be `torch.float`.
+        stride: the stride of the convolving kernel. Can be a single number or a
+          tuple `(sW,)`. Default: 1
+        padding: implicit paddings on both sides of the input. Can be a
+          single number or a tuple `(padW,)`. Default: 0
+        dilation: the spacing between kernel elements. Can be a single number or
+          a tuple `(dW,)`. Default: 1
+        groups: split input into groups, :math:`\text{in\_channels}` should be divisible by the
+          number of groups. Default: 1
+        padding_mode: the padding mode to use. Only "zeros" is supported for quantized convolution at the moment. Default: "zeros"
+        scale: quantization scale for the output. Default: 1.0
+        zero_point: quantization zero_point for the output. Default: 0
+        dtype: quantization data type to use. Default: ``torch.quint8``
+    """
+    stride = _pair_from_first(stride)
+    padding = _pair_from_first(padding)
+    dilation = _pair_from_first(dilation)
+
+    packed_params = torch.ops.quantized.conv1d_prepack_mkldnn(

```

Comment:
running prepacking in runtime would be slower, do you have plans to run the prepacking beforehand?

Reply:
I think weight is always prepacked at runtime for functional conv/linear and weight is prepacked beforehand for conv/linear modules, right?

Reply:
I see, that sounds good I think, but ideally we can use torch.ops.quantized.conv1d directly in the model and prepack the weights beforehand, instead of using functionals